packages/stats/src/index.ts

```diff
@@ -0,0 +1,159 @@
+import * as Parser from '@tracespace/parser'
+
+export interface DrillUsage {
+  diameter: number
+  count: number
+}
+
+export interface DrillStats {
+  drillHits: DrillUsage[]
+  drillRoutes: DrillUsage[]
+  totalDrills: number
+  totalRoutes: number
+  minDrillSize: number
+  maxDrillSize: number
+}
+
+/**
+ * Method to collect drill stats of parsed drill files.
+ *
+ * @example
+ * ```ts
+ * import {createParser} from '@tracespace/parser'
+ * import {collectDrillStats} from '@tracespace/stats'
+ *
+ * const parser = createParser()
+ *
+ * parser.feed(...)
+ *
+ * const tree = parser.results()
+ * const stats = collectDrillStats([tree])
+ * ```
+ *
+ * @category Stats
+ */
+export function collectDrillStats(trees: Parser.Root[]): DrillStats {
+  const state: DrillStatsState = {
+    usedTools: {},
+    drillsPerTool: {},
+    routesPerTool: {},
+    minDrillSize: null,
+    maxDrillSize: null,
+  }
+
+  for (const tree of trees) {
+    if (tree.filetype !== Parser.DRILL) {
+      continue
+    }
+
+    _updateDrillStats(state, tree)
```

Comment:
Functions that mutate an input parameter to do their main work are usually a bad idea. I would solve this problem a different way, perhaps with a function that takes a single tree and returns the stats and another function that combines the stats.=

Reply:
I agree that `DrillStatsState` doesn't lend itself to merging, but it's also a completely internal interface. If `DrillStatsState`'s shape is such that it leads to doing things the way `_updateDrillStats` is currently written, I think `DrillStatsState` should be changed